samples/algorithms/sudoku-grover/Sudoku.qs

```diff
@@ -0,0 +1,203 @@
+        } elif (size != 4) {
+            fail $"Cannot set size {size}: only a grid size of 4x4 or 9x9 is supported";
+        }
+        let numIterations = NIterations(bitsPerColor * numVertices);
+        Message($"Running Quantum test with #Vertex = {numVertices}");
+        Message($"   Bits Per Color = {bitsPerColor}");
+        Message($"   emptySquareEdges = {emptySquareEdges}");
+        Message($"   startingNumberConstraints = {startingNumberConstraints}");
+        Message($"   estimated #iterations needed = {numIterations}");
+        Message($"   size of Sudoku grid = {size}x{size}");
+        let coloring = FindColorsWithGrover(numVertices, bitsPerColor, numIterations, oracle);
+
+        Message($"Got sudoku solution: {coloring}");
+        if (IsSudokuSolutionValid(size, emptySquareEdges, startingNumberConstraints, coloring)) {
+           Message($"Got valid sudoku solution: {coloring}");
+           return (true, coloring);
+        } else {
+           Message($"Got invalid sudoku solution: {coloring}");
+           return (false, coloring);
+        }
+    }
+
+    /// # Summary
+    /// Estimate the number of interations required for solution
+    ///
+    /// # Input
+    /// ## nQubits
+    /// The number of qubits being used
+    function NIterations(nQubits : Int) : Int {
+        let nItems = 1 <<< nQubits; // 2^numQubits
+        // compute number of iterations:
+        let angle = ArcSin(1. / Sqrt(IntAsDouble(nItems)));
+        let nIterations = Round(0.25 * PI() / angle - 0.5);
+        return nIterations;
+    }
+
+    /// # Summary
+    /// Check if the colors/numbers found for each empty square are in the correct 
+    /// range (e.g. <9 for a 9x9 puzzle) and satisfy all edge/starting number constraints
+    /// 
+    /// # Input
+    /// ## size
+    /// The size of the puzzle. 4 for 4x4 grid, 9 for 9x9 grid
+    /// ## edges
+    /// The traditional edges passed to the graph coloring algorithm which, 
+    /// in our case, are empty puzzle squares.
+    /// These edges define any "same row", "same column", "same sub-grid" 
+    /// relationships between empty cells
+    /// Look at the README.md sample output to see examples of what this is 
+    /// for different sample puzzles
+    /// ## startingNumberConstraints
+    /// The constraints on the empty squares due to numbers already in the 
+    /// puzzle when we start. Look at the README.md sample output to see 
+    /// examples of what this is for different sample puzzles
+    /// ## colors
+    /// an Int array of numbers for each empty square i.e. the puzzle solution
+    /// 
+    /// # Output
+    /// A boolean value of true if the colors found satisfy all the solution requirements
+    function IsSudokuSolutionValid (size : Int, edges : (Int, Int)[], 
+        startingNumberConstraints : (Int, Int)[], colors : Int[]) : Bool {
+        for (color in colors) {
+            if (color >= size) {
+                return false;
+            }
+        }
```

Comment:
`Any` and [`GreaterThanOrEqualI`](https://docs.microsoft.com/en-us/qsharp/api/qsharp/microsoft.quantum.logical.greaterthanorequali) may be helpful here.
```suggestion
        if (Any(GreaterThanOrEqualI(_, size), colors)) { return false; }
```